docs/book/src/developer/core/tilt.md

```diff
@@ -8,26 +8,154 @@ workflow that offers easy deployments and rapid iterative builds.
 ## Prerequisites
 
 1. [Docker](https://docs.docker.com/install/): v19.03 or newer (on MacOS e.g. via [Lima](https://github.com/lima-vm/lima))
-2. [kind](https://kind.sigs.k8s.io): v0.25.0 or newer
-3. [Tilt](https://docs.tilt.dev/install.html): v0.30.8 or newer
-4. [kustomize](https://github.com/kubernetes-sigs/kustomize): provided via `make kustomize`
-5. [envsubst](https://github.com/drone/envsubst): provided via `make envsubst`
-6. [helm](https://github.com/helm/helm): v3.7.1 or newer
-7. Clone the [Cluster API](https://github.com/kubernetes-sigs/cluster-api) repository
+1. [kind](https://kind.sigs.k8s.io): v0.25.0 or newer
+1. [Tilt](https://docs.tilt.dev/install.html): v0.30.8 or newer
+1. [kustomize](https://github.com/kubernetes-sigs/kustomize): provided via `make kustomize`
+1. [envsubst](https://github.com/drone/envsubst): provided via `make envsubst`
+1. [helm](https://github.com/helm/helm): v3.7.1 or newer
+1. [virtctl](https://kubevirt.io/user-guide/user_workloads/virtctl_client_tool/): v1.4.0 or newer (required for KubeVirt only)
+1. [ctlptl](https://github.com/tilt-dev/ctlptl): v0.8.37 or newer (required for KubeVirt only)
+1. Clone the [Cluster API](https://github.com/kubernetes-sigs/cluster-api) repository
    locally
-8. Clone the provider(s) you want to deploy locally as well
+1. Clone the provider(s) you want to deploy locally as well
 
 ## Getting started
 
 ### Create a kind cluster
-A script to create a KIND cluster along with a local Docker registry and the correct mounts to run CAPD is included in the hack/ folder.
+
+The following CAPI infrastructure providers are suitable for local development:
+
+- [CAPD](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/README.md) - uses Docker containers as workload cluster nodes
+- [CAPK](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt) - uses KubeVirt VMs as workload cluster nodes
+
+CAPD is the default as it's more lightweight and requires less setup. KubeVirt is useful when
+Docker isn't suitable for whatever reason.
+
+{{#tabs name:"tab-management-cluster-creation" tabs:"Docker,KubeVirt"}}
+{{#tab Docker}}
+
+A script to create a kind cluster along with a local Docker registry and the correct mounts to run CAPD is included in the hack/ folder.
 
 To create a pre-configured cluster run:
 
 ```bash
 ./hack/kind-install-for-capd.sh
 ```
 
+{{#/tab }}
+{{#tab KubeVirt}}
+
+Create a ctlptl configuration for a kind cluster with a local container registry:
```

Comment:
./hack/kind-install-for-capd.sh already creates a kind cluster with a registry, the only thing it does not support is to disable CNI. 

Have you considered to improve ./hack/kind-install-for-capd.sh so we can have a consistent UX (re-use the same registry which persist across test runs, avoid ctlptl + related config file)

Reply:
Can we keep image registry handling as it is given that it is orthogonal to this PR?

I don't remember exact details, but it was intentional to leave the registry around. 
Probably something about avoiding to reload images e.g. in case you are testing on a fixed tag name, but I don't have time to look this up ATM